lib/puppet/datatypes.rb

```diff
@@ -207,7 +207,7 @@ def implementation_class(ruby_class)
     end
 
     def load_file(file_name)
-      Puppet::Util::Autoload.load_file(file_name, nil)
+      Puppet::Util::Autoload.load_file(file_name, Puppet.lookup(:current_environment))
```

Comment:
@hlindberg I think previously the master could end up loading datatypes from the master's configured environment, instead of the node's environment.

Reply:
I took a look at the autoloader and it looks like giving it a `nil` environment would mean using the default and not the current environment. The intention in the data types API is that it should be the current environment - so the change here looks correct.

Reply:
The problem here would only affect use of custom data type with custom ruby implementation and I think the only use of that right now is in Bolt.